src/main/scala/mesosphere/marathon/core/leadership/impl/LeadershipCoordinationActor.scala

```diff
@@ -0,0 +1,88 @@
+package mesosphere.marathon.core.leadership.impl
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Props, Stash, Status, Terminated }
+import akka.event.LoggingReceive
+import mesosphere.marathon.core.leadership.PreparationMessages
+import mesosphere.marathon.core.leadership.impl.WhenLeaderActor.{ Stopped, Stop }
+
+private[leadership] object LeadershipCoordinationActor {
+  def props(whenLeaderActors: Set[ActorRef]): Props = {
+    Props(new LeadershipCoordinationActor(whenLeaderActors))
+  }
+}
+
+private class LeadershipCoordinationActor(var whenLeaderActors: Set[ActorRef])
+    extends Actor with ActorLogging with Stash {
+
+  override def preStart(): Unit = {
+    whenLeaderActors.foreach(context.watch)
+
+    super.preStart()
+  }
+
+  override def receive: Receive = suspended
+
+  private[impl] def suspended: Receive = {
+    log.info("All actors suspended:\n{}", whenLeaderActors.map(actorRef => s"* $actorRef").mkString("\n"))
+
+    LoggingReceive.withLabel("suspended") {
+      case Terminated(actorRef) =>
```

Comment:
In case of a terminated WhenLeaderActor the instance is just removed. 
Is'nt that a problema and should be handled more carefully?